### PR TITLE
[FIX] [16.0] base_tier_validation: Tier Validation Exception Groups

### PR DESCRIPTION
**Pull request by Shide**
Add groups to Tier Validation Exceptions.

This will allow to set some fields only to certain group of users.

This PR is a simplification of https://github.com/OCA/server-ux/pull/913 and https://github.com/OCA/sale-workflow/pull/3222

https://www.loom.com/share/543a78d46cef4323b30a56cc5ad7329f

MT-6238 @moduon @rafaelbn @yajo @LoisRForgeFlow please review if you want :)

### CODE REVIEW

**Review by Gelojr — APPROVED** [review]
LGTM @Shide

File: base_tier_validation/models/tier_validation.py [hunk 1]
```diff
@@ -246,18 +246,22 @@ def _get_validation_exceptions(self, extra_domain=None, add_base_exceptions=True
...
             self.env["tier.validation.exception"]
+            .sudo()
             .search(
                 [
                     ("model_name", "=", self._name),
                     ("company_id", "in", [False] + self.env.company.ids),
                     *(extra_domain or []),
+                    "|",
+                    ("group_ids", "in", self.env.user.groups_id.ids),
+                    ("group_ids", "=", False),
```
Comment (yajo) [inline base_tier_validation/models/tier_validation.py]: issue: for security, always append extra domains at the end:
```suggestion
                    "|",
                    ("group_ids", "in", self.env.user.groups_id.ids),
                    ("group_ids", "=", False),
                    *(extra_domain or []),
```

This avoids the case of dangling operands that alter the meaning of the rest of the domain.

File: base_tier_validation/templates/tier_validation_templates.xml [hunk 2]
```diff
@@ -5,80 +5,81 @@
...
+                ('rejected','=',True),('#{state_field}', '#{state_operator}', #{state_value})]}"
                 type="object"
             />
             <button
                 name="restart_validation"
                 string="Restart Validation"
-                t-attf-attrs="{'invisible': ['|',('review_ids', '=', []),('#{state_field}', '#{state_operator}', #{state_value})]}"
+                t-attf-attrs="{'invisible': ['|',
+                ('review_ids', '=', []),
+                ('#{state_field}', '#{state_operator}', #{state_value})]}"
```
Comment (yajo) [inline base_tier_validation/templates/tier_validation_templates.xml:18]: suggestion: these style changes could be reverted.

**Review by yajo — APPROVED** [review]

**Review by yajo — CHANGES_REQUESTED** [review]
Oops sorry wrong button... [the issue](https://github.com/OCA/server-ux/pull/915#discussion_r1670064291) should be solved at least.

File: base_tier_validation/templates/tier_validation_templates.xml (hunk 2 above)
Comment (Shide) [reply base_tier_validation/templates/tier_validation_templates.xml:18]: Helps to read the code better

**Comment by Shide** [comment]
@LoisRForgeFlow what do you think with these changes?

**Review by yajo — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Functional tests and code review. Works as expected, nice improvement :+1:

**Pull request merged**